### PR TITLE
add wasm build option for filewriter flags[skip changelog]

### PR DESCRIPTION
**Pull request by mar1n3r0**

### CODE REVIEW

**Review by Jorropo — CHANGES_REQUESTED** [review]
This looks wrong to me, right now it is a list of compatible GOOSes but you are adding a GOARCH.

If windows added support for running WASM binaries this would be incorrect.

What do you think of:
> ```
> //go:build darwin || linux || netbsd || openbsd || freebsd || dragonfly || js || wasip1
> ```
?

**Comment by mar1n3r0** [comment]
Yep, totally makes sense.

File: files/filewriter_unix.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//go:build darwin || linux || netbsd || openbsd || freebsd || dragonfly
+//go:build darwin || linux || netbsd || openbsd || freebsd || dragonfly || wasm
```
Comment (Jorropo) [inline files/filewriter_unix.go]:
```suggestion
//go:build darwin || linux || netbsd || openbsd || freebsd || dragonfly || js || wasip1
```

File: files/filewriter_unix_test.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-//go:build darwin || linux || netbsd || openbsd
+//go:build darwin || linux || netbsd || openbsd || freebsd || dragonfly || wasm
```
Comment (Jorropo) [inline files/filewriter_unix_test.go]:
```suggestion
//go:build darwin || linux || netbsd || openbsd || freebsd || dragonfly || js || wasip1
```

**Review by Jorropo — APPROVED** [review]
LGTM thx

**Comment by Jorropo** [comment]
I can't apply the suggestions, could you do it please ?

**Comment by mar1n3r0** [comment]
> I can't apply the suggestions, could you do it please ?

All set.

File: files/filewriter_unix_test.go (hunk 2 above)
Comment (hacdias) [reply files/filewriter_unix_test.go]: @mar1n3r0 this suggestion is still missing, can you pls apply it?

**Comment by mar1n3r0** [comment]
Sorry, forgot about that. Done.

**Pull request merged**